final_task/rss_reader/cache.py

```diff
@@ -0,0 +1,105 @@
+import pickle
+import logging
+from os import path, makedirs
+from dateutil.parser import parse
+
+from exceptions import CacheNotFoundError
+
+
+# cache storage directory
+CACHE_PATH = '/tmp/rss_reader'
+
+
+def save_cache(news_articles, url):
+    """Save news_articles to cache"""
+    # cache {date: {url: set(news_articles)}}
+    cache = format_cache(news_articles, url)
+    for date in cache:
+        # cache_from_file {url: set(news_articles)}
+        cache_from_file = read_from_file(date)
+        result_cache = update_cache(cache_from_file, cache, date, url)
+        logging.info(f'Saving cache to {CACHE_PATH}/cache/{date}.cache')
+        write_to_file(result_cache, date)
+    logging.info('Cache is saved')
+
+
+def read_cache(date, url, limit):
+    # get cache for this url from file
+    logging.info(f'Loading cache from {CACHE_PATH}/cache/{date}.cache')
+    # cache {url: set(news_articles)}
+    cache = read_from_file(date)
+    news_articles = []
+    if cache:
+        if url == 'ALL':
+            # get all news for this date
+            for outlett in cache:
+                for article in cache[outlett]:
+                    if len(news_articles) == limit:
+                        break
+                    news_articles.append(article)
+        else:
+            # get news only for these url and date
+            if cache.get(url):
+                for article in cache[url]:
+                    if len(news_articles) == limit:
+                        break
```

Comment:
It is better to use slices (`[:]`) for this instead of comparing articles length to the limit on each iteration.

Reply:
Thanks. I fixed it.